typing/rec_check.ml

```diff
@@ -909,6 +909,7 @@ and path : Path.t -> term_judg =
           path f << Dereference;
           path p << Dereference;
         ]
+    | Path.Pextra_ty _ -> assert false (* Only in type paths *)
```

Comment:
We looked at the code and currently the comment is correct, we believe that `path pth` is never called on the path `pth` to a type. So in particular `Pextra_ty` should never show up.

But: given that the code is erring on the safe/conservative side of calling `path` on all paths showing up in the typedtree, it's plausible that there would be calls on such paths in the future, and `assert false` is a bit harsh.

What could we do instead?
- `Path.Pextra_ty (p, _extra) -> path p` : conservative solution (might reject some programs that could be accepted)
- `empty`: liberal solution (probably safe, type paths should not imply any runtime value dependency; but if we are wrong, this might accept some programs that should be rejected)

My intuition is that the conservative solution is best.

Reply:
Gone with conservative solution, as you pointed at #11568